### DISCUSSION TITLE
How to use this lib for zlib?

### DISCUSSION THREAD

**Discussion by StefanOltmann** [korlibs/korge#2087] [Q&A] [answered]
Since https://github.com/korlibs-archive/kzlib I wonder how I can use this lib for zlib compression. 

My lib now uses the pako.js for zlib, but I would prefer a purely Kotlin native way.

See https://github.com/Ashampoo/kim/blob/b3dac7e88bb03636b68099ed065a81bd0e5e3d08/build.gradle.kts#L288

Btw: Why was kzlib deprecated?

**Comment by soywiz** [comment, marked as answer]
I see.

kzlib was archived, because it was reimplemented without licensing issues into korlibs-core to simplify licensing. Since uncompressing is simpler.
Also maintaining that many repos was really costly in terms of time, so I unified libraries and repos as much as possible to save my own time.

You can include `korlibs-core` that includes the io package. Otherwise, you can copy those sources from kzlib into your library with the zlib license. Most of the code is common except for the arraycopy, but with recent Kotlin versions yo should be able to use the copyInto function and all the code should be common: https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/copy-into.html

Hope it helps!
